src/nupic/support/pymysqlhelpers.py

```diff
@@ -113,28 +113,33 @@ def retrySQL(timeoutSec=60*5, getLoggerCallback=logging.getLogger):
     def jobInfo(self, jobID):
         ...
   """
-  
+
+  if getLoggerCallback:
```

Comment:
If the default value is None, then this check should explicitly check `if getLoggerCallback is not None`, instead of relying on coercion to bool. This is because in python objects (including callables) may override the bool coercion to attain some application-specific esoteric meaning (e.g., evaluates to false if it's never been called). So, if `None` is the documented default value designator, then check explicitly for `None`.